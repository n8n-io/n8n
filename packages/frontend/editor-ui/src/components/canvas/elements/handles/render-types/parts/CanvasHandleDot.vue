<script lang="ts" setup>
withDefaults(
	defineProps<{
		handleClasses?: string;
	}>(),
	{
		handleClasses: undefined,
	},
);
</script>

<template>
	<div :class="[$style.dot, handleClasses]" />
</template>

<style lang="scss" module>
.dot {
	width: var(--handle--indicator--width);
	height: var(--handle--indicator--height);
	border-radius: 50%;
	background: var(--color-foreground-xdark);

	&:hover {
		background: var(--color-primary);
	}
}
</style>
