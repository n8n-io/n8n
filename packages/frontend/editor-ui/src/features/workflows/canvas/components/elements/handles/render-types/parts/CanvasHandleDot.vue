<script lang="ts" setup>
withDefaults(
	defineProps<{
		handleClasses?: string;
	}>(),
	{
		handleClasses: undefined,
	},
);
</script>

<template>
	<div :class="[$style.dot, handleClasses]" />
</template>

<style lang="scss" module>
.dot {
	width: var(--handle--indicator--width);
	height: var(--handle--indicator--height);
	border-radius: 50%;
	background: var(--color--foreground--shade-2);

	&:hover {
		background: var(--color--primary);
	}
}
</style>
