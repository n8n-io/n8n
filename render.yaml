services:
  - type: web
    name: luci-n8n-agent
    env: docker
    plan: free
    branch: master
    autoDeploy: true
    buildCommand: ""
    startCommand: "n8n start"
    envVars:
      - key: N8N_BASIC_AUTH_ACTIVE
        value: "true"
      - key: N8N_BASIC_AUTH_USER
        value: "admin"
      - key: N8N_BASIC_AUTH_PASSWORD
        value: "admin"
      - key: N8N_HOST
        value: "0.0.0.0"
      - key: N8N_PORT
        value: "5678"
      - key: WEBHOOK_URL
        value: https://luci-n8n-agent.onrender.com/
