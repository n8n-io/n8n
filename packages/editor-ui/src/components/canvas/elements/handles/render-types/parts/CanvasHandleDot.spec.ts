import CanvasHandleDot from './CanvasHandleDot.vue';
import { createComponentRenderer } from '@/__tests__/render';

const renderComponent = createComponentRenderer(CanvasHandleDot, {});

describe('CanvasHandleDot', () => {
	it('should render with default props', () => {
		const { html } = renderComponent();

		expect(html()).toMatchSnapshot();
	});

	it('should apply `handleClasses` prop correctly', () => {
		const customClass = 'custom-handle-class';
		const wrapper = renderComponent({
			props: { handleClasses: customClass },
		});

		expect(wrapper.container.querySelector(`.${customClass}`)).toBeTruthy();
	});
});
