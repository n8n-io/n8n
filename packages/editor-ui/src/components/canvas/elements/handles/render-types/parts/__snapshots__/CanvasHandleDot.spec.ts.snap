// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CanvasHandleDot > should render with default props 1`] = `"<div class="dot"></div>"`;
